include/deal.II/base/mpi_noncontiguous_partitioner.templates.h

```diff
@@ -310,6 +310,8 @@ namespace Utilities
       // post recv
       for (types::global_dof_index i = 0; i < recv_ranks.size(); i++)
         {
+          AssertIndexRange(i + 1, recv_ptr.size());
+          AssertIndexRange(i + send_ranks.size(), requests.size());
```

Comment:
It should be sufficient to only check for the last index, isn't it?
```suggestion
          AssertIndexRange(recv_ranks.size(), recv_ptr.size());
          AssertIndexRange(recv_ranks.size() + send_ranks.size(), requests.size() + 1);
```
and moving out of the loop; the same for the cases below.

Reply:
Yes - good point.